addons/html_editor/static/tests/toolbar.test.js

```diff
@@ -1110,3 +1110,30 @@ describe("toolbar open and close on user interaction", () => {
         });
     });
 });
+
+describe.tags("desktop");
+describe("toolbar visibility on contenteditable false elements", () => {
+    test("should not open toolbar when selected element is contenteditble false", async () => {
+        await setupEditor('<div contenteditable="false"><p>a[bc<strong>def</strong>gh]i</p></div>');
+        await animationFrame();
+        expect(".o-we-toolbar").toHaveCount(0);
+    });
+
+    test("should open the toolbar when the selected t-out is contenteditable false", async () => {
+        await setupEditor('<div contenteditable="false"><t t-out="">[abc]</t></div>');
+        await animationFrame();
+        expect(".o-we-toolbar").toHaveCount(1);
+    });
+
+    test("should open the toolbar when the selected t-field is contenteditable false", async () => {
+        await setupEditor('<div contenteditable="false"><t t-field="">[abc]</t></div>');
+        await animationFrame();
+        expect(".o-we-toolbar").toHaveCount(1);
+    });
+
+    test("should open the toolbar when the selected t-esc is contenteditable=false", async () => {
+        await setupEditor('<div contenteditable="false"><t t-esc="">[abc]</t></div>');
+        await animationFrame();
+        expect(".o-we-toolbar").toHaveCount(1);
+    });
+});
```

Comment:
@Zynton Could you have a look at those please? I want to be sure it won't break your caption feature and I don't remember exactly the details of what is editable, what is selected etc

Reply:
I'm a bit confused with this. Are `t-esc` and `t-field` automatically `contenteditable="true"`? Do we want to allow the toolbar in them because we know they will be later replaced with elements that have `contenteditable="true"` as attribute?

Reply:
It's about studio, see the discussion above. But now that I have re-read it, I think the tests here are false. @dhba-odoo In the studio case, isn't the container of the t-esc _editable_? I think that was the point, that even though none of the nodes selected were editable, the selection itself rested inside a contenteditable=true zone so it should be considered editable, see what I mean?

So, like: `<div contenteditable="true">[<t t-esc="" contenteditable="false">abc</t>]</div>` should show the toolbar.